===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(rec_db_merge_records:new_name)) at 17 (line 1015, column 14) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/recutils-1.8/src/rec-db.c
+++ b/table1/benchmarks/recutils-1.8/src/rec-db.c
@@ -1014,6 +1014,7 @@ rec_db_merge_records (rec_record_t record1,
 
         if (!rec_field_set_name (new_field, new_name))
           {
+            free(new_name);
             /* Out of memory.  */
             return NULL;
           }
